### PR TITLE
cli-ux migration with @oclif/core package - CS-26598

### PR DESCRIPTION
**Pull request by antonyagustine**
CLI-UX package migration with @oclif/core package

### CODE REVIEW

File: packages/contentstack-auth/src/commands/auth/tokens/index.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+import { logger, cliux, ux, configHandler } from '@contentstack/cli-utilities';
 
 export default class TokensListCommand extends Command {
   private readonly parse: Function;
   static run;
   static description = 'Lists all existing tokens added to the session';
   static aliases = ['tokens'];
   static examples = ['$ csdx auth:tokens'];
-  static flags = cli.table.flags(); // use the cli table flags as it displays tokens in table
+  static flags = ux.table.flags(); // use the cli table flags as it displays tokens in table
```
Comment (shafeeqd959) [inline packages/contentstack-auth/src/commands/auth/tokens/index.ts:10]: do we need separate interface for table flags, cant we merge it with cliux

File: packages/contentstack-bootstrap/src/bootstrap/interactive.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
-import * as inquirer from 'inquirer';
-import cli from 'cli-ux';
 import * as path from 'path';
+import * as inquirer from 'inquirer';
+import { ux as cli } from '@contentstack/cli-utilities';
+
```
Comment (shafeeqd959) [inline packages/contentstack-bootstrap/src/bootstrap/interactive.ts:4]: should be exported as cliux

File: packages/contentstack-command/src/index.ts [hunk 3]
```diff
@@ -45,6 +45,7 @@ abstract class ContentstackCommand extends Command {
 
   get region() {
     if (this._region) return this._region;
+    console.log(configHandler, 'configHandler')
     this._region = configHandler.get('region');
```
Comment (shafeeqd959) [inline packages/contentstack-command/src/index.ts:49]: remove this

File: packages/contentstack-seed/src/typings/@contentstack/cli-utilities.d.ts [hunk 4]
```diff
@@ -1,5 +1,43 @@
 declare module '@contentstack/cli-utilities' {
```
Comment (shafeeqd959) [inline packages/contentstack-seed/src/typings/@contentstack/cli-utilities.d.ts:1]: do we need to declare the types in this package

File: packages/contentstack-utilities/src/cli-ux.ts [hunk 5]
```diff
@@ -41,15 +42,23 @@ class CLIInterface {
...
     inquirePayload.message = messageHandler.parse(inquirePayload.message);
     const result = await inquirer.prompt(inquirePayload as inquirer.QuestionCollection<T>);
+
     return result[inquirePayload.name] as T;
   }
 }
 
 export default new CLIInterface();
+
+export * as CliUx from '@oclif/core/lib/cli-ux'
```
Comment (shafeeqd959) [inline packages/contentstack-utilities/src/cli-ux.ts:63]: we should be exporting only CLIInterface instance

File: packages/contentstack-utilities/src/config-handler.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ class Config {
   private config: Conf;
 
   constructor() {
-    this.init()
+    // this.init()
```
Comment (shafeeqd959) [inline packages/contentstack-utilities/src/config-handler.ts:10]: uncomment this

File: packages/contentstack/src/utils/context-handler.ts [hunk 7]
```diff
@@ -25,7 +26,7 @@ export default class CsdxContext {
       authtoken: configHandler.get('authtoken'),
       email: configHandler.get('email'),
     };
-    this.config = { ...config };
+    this.config = { ...config } // configHandler.init();
```
Comment (shafeeqd959) [inline packages/contentstack/src/utils/context-handler.ts:29]: revert configuration changes

File: packages/contentstack-auth/src/commands/auth/tokens/index.ts (hunk 1 above)
Comment (antonyagustine) [reply packages/contentstack-auth/src/commands/auth/tokens/index.ts]: Add computer property in CLIInterface and reading from there.

File: packages/contentstack-command/src/index.ts (hunk 3 above)
Comment (antonyagustine) [reply packages/contentstack-command/src/index.ts]: Removed log

File: packages/contentstack-seed/src/typings/@contentstack/cli-utilities.d.ts (hunk 4 above)
Comment (antonyagustine) [reply packages/contentstack-seed/src/typings/@contentstack/cli-utilities.d.ts]: removed type definition module from all modules and added type index in the package.json

File: packages/contentstack-utilities/src/cli-ux.ts (hunk 5 above)
Comment (antonyagustine) [reply packages/contentstack-utilities/src/cli-ux.ts]: removed other export. Using only CLIInterface

File: packages/contentstack-utilities/src/config-handler.ts (hunk 6 above)
Comment (antonyagustine) [reply packages/contentstack-utilities/src/config-handler.ts]: reverted commented code

File: packages/contentstack/src/utils/context-handler.ts (hunk 7 above)
Comment (antonyagustine) [reply packages/contentstack/src/utils/context-handler.ts:29]: reverted back to old code.

**Pull request merged**